frame/message-queue/src/benchmarking.rs

```diff
@@ -274,6 +274,78 @@ mod benchmarks {
 		assert!(Pages::<T>::contains_key(&origin, 0), "Page must be updated");
 	}
 
+	#[benchmark]
+	fn discard_overweight_page_removed() -> Result<(), BenchmarkError> {
+		let queue: MessageOriginOf<T> = 0.into();
+		let origin = T::DiscardOverweightOrigin::try_successful_origin(&queue)
+			.map_err(|_| BenchmarkError::Weightless)?;
+		let (mut page, msgs) = full_page::<T>();
+		// Skip all messages.
+		for _ in 1..msgs {
+			page.skip_first(true);
+		}
+		page.skip_first(false);
+		let book = book_for::<T>(&page);
+		Pages::<T>::insert(&queue, 0, &page);
+		BookStateFor::<T>::insert(&queue, &book);
+		// Remove the last message since `peek_index` has linear complexity.
+		let message_index = ((msgs - 1) as u32).into();
+
+		#[block]
+		{
+			MessageQueue::<T>::discard_overweight(origin, 0u32.into(), 0u32, message_index)
+				.unwrap();
+		}
```

Comment:
just want to double check we are actually stuck using `#[block]` and not the shorter syntax here? Anything we can do in benchmarking v2 to improve this?

Reply:
Ah right. I think this MR predates the `as T::RuntimeOrigin` syntax.  
It works now with that :smile: